### ISSUE TITLE
Graphical monitor calibration to address fractional scaling

### ISSUE THREAD

**Issue by mmahmoudian** [flameshot-org/flameshot#2327]
### Feature Description

This feature requests has been briefly discussed in https://github.com/flameshot-org/flameshot/issues/1386#issuecomment-991652151

Considering that the 4k monitors are getting more and more adopted, and more people will end up using fractional scaling, I think it worth the effort to implement a "calibration" mechanism. The following is my suggestion:

We display an overlay and ask the user to scroll mouse while hovering on different monitors (or press keyboard shortcuts) to align certain calibration lines. For instance assume we use the same overlay as `flameshot gui` but instead of being able to select an area, it has some lines crossing each other at the center basically a combination of + and x (a star that the lines go to infinity).

Typically users use the same monitor setup, so we can save the offsets (or perhaps coefficients) in the config file at a designated section.

**Comment by borgmanJeremy** [comment]
@mmahmoudian Any thoughts on how this should work for laptop users that constantly have new monitor configurations?

**Comment by mmahmoudian** [comment]
Good question, but tbh I've never seen anyone set fractional scaling on a temp monitor, TV or projector. Typically either it is always the same monitor from the same port (e.g monitor is at workplace desk and the user take the laptop home or out of office every day), or it is a random temporary display on which usually the user doesn't use a screenshot tool. 

So I still believe this is still a good solution. Also if using the calibration is easy fast and intuitive enough, I don't see why the user should hesitate in calibrating the new display on the fly.

I don't think there is a universal solution out there. Do you have any ideas/suggestions that would work? I'm more than willing to explore other avenues 🤓

**Comment by borgmanJeremy** [comment]
I agree it's still a good idea. The laptop problem randomly popped into my head. I can't think of a good workaround yet. Maybe hash the display config and use that as a key? Probably overkill.

**Comment by mmahmoudian** [comment]
The hash to be used as ID sounds interesting, although perhaps too much effort for too little gain. We can implem@ this without the monitor identification (hash or any other method) and see how much of our users will contact us for this monitor identification feature.

**Comment by borgmanJeremy** [comment]
@mmahmoudian I read through our issues again, and I wonder if this is actually needed. It seems like setting the environmental variable solves the problem for everyone?
```
QT_SCREEN_SCALE_FACTORS
```

**Comment by mmahmoudian** [comment]
@borgmanJeremy I have already added that to our documentation:
https://flameshot.org/docs/guide/wayland-help/#kde

but it seems there are two values that have been reported that works (`QT_SCREEN_SCALE_FACTORS="1;1"` and `QT_SCREEN_SCALE_FACTORS=""`) and I don't know which one is the correct one, plus it seems this works for Wayland people and I'm not sure if X11 is solved by this.

If simply setting one variable in the environment solves all issues, we should definitely avoid implementing a calibration mechanism. Do you know how this can be tested/confirmed?

**Comment by ihipop** [comment]
> QT_SCREEN_SCALE_FACTORS

disable `QT_SCREEN_SCALE_FACTORS` works,but it just works instead of works well
it will cause flameshot to show a not scaled UI widget, you have to set a bigger font DPI to make it looks  like (some kind of ) good, but font DPI  is a global setting

`calibration` will make flameshot  to show a proper scaled UI,and best UI status scaled

**Comment by mmahmoudian** [comment]
Fractional scaling was addressed in #4498. I'll close this.
